dali/pipeline/data/buffer.h

```diff
@@ -88,6 +88,60 @@ inline bool set_deletion_order(const std::shared_ptr<T> &ptr, AccessOrder order)
   return set_deletion_order(std::static_pointer_cast<void>(ptr), order);
 }
 
+/**
+ * @brief Checks whether the underlying managed object of the shared pointers is the same.
+ *
+ * This function may return true for two pointers that do not compare equal if they were created
+ * as aliasing different shared pointers.
+ */
+template <typename T, typename U>
+bool same_managed_object(const std::shared_ptr<T> &a, const std::shared_ptr<U> &b) {
+  return !a.owner_before(b) && !b.owner_before(a);
+}
+
+template <typename T, typename U>
+bool same_managed_object(const std::shared_ptr<T> &a, const std::weak_ptr<U> &b) {
+  return !a.owner_before(b) && !b.owner_before(a);
+}
+
+template <typename T, typename U>
+bool same_managed_object(const std::weak_ptr<T> &a, const std::shared_ptr<U> &b) {
+  return !a.owner_before(b) && !b.owner_before(a);
```

Comment:
Does it make sense to express it using L103
```suggestion
  return same_managed_object(b, a);
```

Reply:
I don't think it's worth it - the savings in the number of letters typed are marginal and readability isn't improved - rather the contrary. Now it's quite obvious that these functions do the same thing.

Reply:
Sure, just an idea.